src/components/Card.js

```diff
@@ -49,17 +89,44 @@ export class Card {
     this._deleteButton = this._cardElement.querySelector(
       ".card__delete-button"
     );
-    this._deleteButton.addEventListener("click", () => {
-      this._deleteCard();
-    });
+    this._deleteButton.addEventListener("click", this._handleDelete);
   }
 
-  _deleteCard() {
+  deleteCard() {
+    console.log("you pressed this delete button");
+    // this._handleDelete.setEventListeners();
+    // this._handleDelete.open();
     this._cardElement.remove();
     this._cardElement = null;
   }
 
+  refreshLikesCount(likesCount) {
+    likesCount < 1
+      ? (this._cardElement.querySelector(".card__likes-counter").textContent =
+          "")
+      : (this._cardElement.querySelector(".card__likes-counter").textContent =
+          likesCount);
+  }
+
   _handleLike() {
-    this._likeButton.classList.toggle("card__like-button_active");
+    if (this._isLiked) {
+      this._handleDislikeCallback();
+      this.dislike();
+    } else if (!this._isLiked) {
+      this._handleLikeCallback();
+      this.addLike();
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.